### ISSUE TITLE
[New Sensor]: CQ60 Chef iQ wireless meat thermometer

### ISSUE THREAD

**Issue by willemcvu** [custom-components/ble_monitor#1279]
### Sensor Description

CQ60 Chef iQ wireless meat thermometer

### Additional information

I started figuring out the advertisement packets (and did a tear-down of the base station) here: https://community.home-assistant.io/t/chef-iq-smart-wireless-meat-thermometer/654248/5

The probe temperature in this log is in the 0x15FFCD0501 messages, and it's the first byte (0x12 and 0x13 aka 18 and 19 deg C in these logs). The rest of the message has not been figured out but the app also reports more data. Here's all the data you can pull from a probe in the app, that likely is encoded in these messages somehow:

- Serial #: PR8P-CT0162342-006186
- MAC address: D9:38:36:2E:33:73
- HW version: v22.0.0
- FW version: v2.0.0
- HVIN: CQ60-QPR-01
- Probe temp
- Ambient temp
- Battery charge level

### BLE advertisements

```shell

2024-01-03 14:44:20.221 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x13\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8d\x11']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:20.727 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x13\xff\xcd\x05\x00@Probe 1\x00\x00\x00\x00\x00C*']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:21.232 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:21.744 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:22.244 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc1\x00\xc8\x00\xc8\x00\xca\x00\xcb\x00\xc1\x00\xe3\xca']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:22.755 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:23.259 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:23.762 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:24.262 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00s\xc3']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:24.766 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:25.269 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:26.285 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:26.791 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x13\xff\xcd\x05\x00@Probe 1\x00\x00\x00\x00\x00C*']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:27.294 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x13\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00qB']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:27.795 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:28.303 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:29.325 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xbf\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xbf\x00\xfa\xcb']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:29.829 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:30.337 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:30.842 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x13\xff\xcd\x05\x00@Probe 1\x00\x00\x00\x00\x00C*']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:31.351 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00s\xc3']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:31.858 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:32.364 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00s\xc3']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:32.873 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:33.380 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xbf\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xbf\x00\xfa\xcb']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:33.882 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:34.890 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xbf\x00\xc8\x00\xc8\x00\xca\x00\xcb\x00\xbf\x00j\xc2']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:35.396 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc0\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc0\x00\x8f\x90']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:35.903 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:36.405 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x12\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00s\xc3']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:36.912 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x13\xff\xcd\x05\x00@Probe 1\x00\x00\x00\x00\x00C*']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:37.418 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x13\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00qB']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:37.928 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\r\xff\xcd\x05\x03@\xd986.3s\x191']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:38.429 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x13\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00qB']local name: CQ60UUID16: None,UUID128: None
2024-01-03 14:44:38.938 INFO (Thread-7) [custom_components.ble_monitor.ble_parser] BLE advertisement received from MAC/UUID d938362e3373: service data: []manufacturer specific data: [b'\x15\xff\xcd\x05\x01@c\x13\xc1\x00\xc9\x00\xc9\x00\xca\x00\xcb\x00\xc1\x00qB']local name: CQ60UUID16: None,UUID128: None

```

**Comment by Ernst79** [comment]
First results
```
{'temperature': 19, 'rssi': -52, 'mac': 'D938362E3373', 'type': 'CQ60', 'packet': 'no packet id', 'firmware': 'Chef iQ', 'data': True}
```

data: `15ffcd0501406313c000c900c900ca00cb00c0008d11`
`15` = length
`ff` = manufacturer data
`cd05` = Manufacturer id `05CD`
`0140` = not sure, possibly a device ID
`63` = 99 = 99% (battery?)
`13` = 19 (temperature?)
`c000` = `00C0` = 192 --> 19.2 (temperature?)
`c900` = `00C9` = 201 --> 20.1 (temperature?)
`c900`= `00C9` = 201 --> 20.1 (temperature?)
`ca00`= `00CA` = 202 --> 20.2 (temperature?)
`cb00`= `00CB` = 203 --> 20.3 (temperature?)
`c000` = `00C0` = 192 --> 19.2 (temperature?)
`8d11` = `118d` = 4493 --> could be humidity 44.93%????

I can make a test version for BLE monitor with all temperatures, but I will have to ask you to figure out which temperature is what.

**Comment by Ernst79** [comment]
I have created 12.8.0-beta to test. I named the sensors as described above. I hope you can tell me which sensor is what? It can also be temperature alarms, or min/max values or ..... Please report back your findings, such that I can name the sensors properly.

**Comment by willemcvu** [comment]
Ok, I have done some experimentation with an ice bath, lighter, and oven. Here is my understanding of what is going on:

![image](https://github.com/custom-components/ble_monitor/assets/32114876/d12ec161-3ff1-4653-b4e6-7433595a5ffa)


The probe has 5 internal temp sensors, with raw values reported:
* Tip
* "Ring 1"
* "Ring 2"
* Ambient
* "Ring 3"

The probe then calculates the following:
* meat internal temperature
  * this is simply the min of tip, ring 1, and ring 2.
* ambient temperature
  * this is simply a repeat of the ambient temp sensor. At all times they were identical.

These map to the fields in your beta version as follows:
* temperature = ring 3 (based on the different format I guess this is an RTD or something compared to whatever the other sensors are, and may also be used for CJC for the thermocouple presumably measuring ambient)
* probe 1 = calculated ambient (am calling this "calculated" based on field order here, with calculated ones first followed by raw data in geographical order)
* probe 2 = calculated internal
* probe 3 = tip
* probe 4 = ring 1
* probe 5 = ring 2
* probe 6 = ambient

This all works _except_ while putting this in the oven, ambient (both types) seems to max out and stop increasing above between 70C and 95C. This is what holding a lighter to the black part of the probe looks like:

![image](https://github.com/custom-components/ble_monitor/assets/32114876/0b4f94e6-c720-4f25-829b-3f5efaa77780)


Confused here, because I can see the bytes going higher than that via BLE scanner app on my phone:

```15FFCD050140571E6F07A000A000AF00FB0079067C44``` -> ```6F07``` -> ```0x076F```=```190.3``` deg

Regarding other data:
* battery: correct (these things have good batteries, it was down to only 94% after ~90 minutes of fooling around with it)
* "humidity": not really sure, but I think this may be a very slow-moving measure of conductivity: ~70% in free air, ~50% in water, ~20% in salt water. Makes sense for seeing moisture level of your meat is I guess?

**Comment by willemcvu** [comment]
ok - actually really not sure on the "humidity" field. I turned the averaging time to 0 and it it's essentially random values. not sure, I think we should ignore it for now.

**Comment by Ernst79** [comment]
Thanks, released as 12.8.0. I added the explanation in the docs and renamed the sensors.
